src/components/Board/TileEditor/TileEditor.component.js

```diff
@@ -47,747 +47,804 @@ import LoadBoardEditor from './LoadBoardEditor/LoadBoardEditor';
-                  </Button>
-                }
-                backButton={
-                  <Button
-                    onClick={this.handleBack}
-                    disabled={this.state.activeStep === 0}
-                  >
-                    <KeyboardArrowLeftIcon />
-                    {intl.formatMessage(messages.back)}
-                  </Button>
-                }
-              />
-            )}
-          </Paper>
-
-          <SymbolSearch
-            open={this.state.isSymbolSearchOpen}
-            autoFill={this.state.autoFill}
-            onChange={this.handleSymbolSearchChange}
-            onClose={this.handleSymbolSearchClose}
-          />
-        </FullScreenDialog>
-      </div>
-    );
-  }
+                    onChange={handleColorChange}
+                    defaultColor={
+                      getCurrentlyEditingTile()
+                        ? getOriginalTileBackground()
+                        : getDefaultColor()
+                    }
+                  />
+                </div>
+                {currentTileProp('type') !== 'board' && (
+                  <div className="TileEditor__voicerecorder">
+                    <FormLabel>
+                      {intl.formatMessage(messages.voiceRecorder)}
+                    </FormLabel>
+                    <PremiumFeature>
+                      <VoiceRecorder
+                        src={currentTileProp('sound')}
+                        onChange={handleSoundChange}
+                      />
+                    </PremiumFeature>
+                  </div>
+                )}
+              </div>
+            </div>
+          </FullScreenDialogContent>
+
+          {tilesToEdit.length > 1 && (
+            <MobileStepper
+              variant="progress"
+              steps={tilesToEdit.length}
+              position="static"
+              activeStep={activeStep}
+              nextButton={
+                <Button
+                  onClick={handleNext}
+                  disabled={activeStep === tilesToEdit.length - 1}
+                >
+                  {intl.formatMessage(messages.next)} <KeyboardArrowRightIcon />
+                </Button>
+              }
+              backButton={
+                <Button onClick={handleBack} disabled={activeStep === 0}>
+                  <KeyboardArrowLeftIcon />
+                  {intl.formatMessage(messages.back)}
+                </Button>
+              }
+            />
+          )}
+        </Paper>
+
+        <SymbolSearch
+          open={isSymbolSearchOpen}
+          autoFill={autoFill}
+          onChange={handleSymbolSearchChange}
+          onClose={handleSymbolSearchClose}
+        />
+      </FullScreenDialog>
+    </div>
+  );
 }
 
+TileEditor.propTypes = {
+  intl: PropTypes.object.isRequired,
+  open: PropTypes.bool,
+  onClose: PropTypes.func.isRequired,
+  editingTiles: PropTypes.array,
+  onEditSubmit: PropTypes.func.isRequired,
+  onAddSubmit: PropTypes.func.isRequired,
```

Comment:
I realized the block comments for these PropTypes are missing, and that `TileEditor.propTypes.intl` needs to be changed back from `PropTypes.object.isRequired` to `intlShape.isRequired`. I'll revert that.